src/queries/query-evaluation-model-in-detail.md

```diff
@@ -76,7 +76,7 @@ executed, no results are cached. But the context already provides access to
 "input" data, i.e. pieces of immutable data that were computed before the
 context was created and that queries can access to do their computations.
 
-As of <!-- date-check --> January 2021, this input data consists mainly of
+As of <!-- date-check --> April 2023, this input data consists mainly of
```

Comment:
I guess we still can say this: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html#method.hir

Reply:
So, does it mean we now have different items other than the ones mentioned here in the query ctx's input data?

Reply:
"It's complicated". I think in practice `crate_for_resolver` is the entry point to the query system, and that's fed here: https://github.com/rust-lang/rust/blob/f98a2718141593fbb8dbad10acc537786d748156/compiler/rustc_interface/src/queries.rs#L239
`krate` comes from `Queries::register_plugins` (which is unrelated to the "query system" 😞) which uses the `Session` and parsed AST as an input.

That said, everything in between `crate_for_resolver` and `all_resolutions` isn't really incremental, like I mentioned in https://github.com/rust-lang/rustc-dev-guide/pull/1662/files/7c94f1af57329b3649355803897590ffa72488ea#r1158541999.